db/memorystore.go

```diff
@@ -170,6 +172,13 @@ func (m *MemoryStore) AddOrder(order *core.Order) (int, error) {
 		return 0, fmt.Errorf("order %q already exists", orderID)
 	}
 
+	var ordersByAccountID []*core.Order
```

Comment:
I think you can get rid of these two explicit `var` declarations and use:

```
if ordersByAccountID, present := m.ordersByAccountID[order.AccountID]; present {
....
}
```

Reply:
I tried that first, but then `ordersByAccountID` won't be available after the `if` block.